byterun/alloc.c

```diff
@@ -46,11 +46,13 @@ CAMLexport value caml_alloc (mlsize_t wosize, tag_t tag)
         Op_val(result)[i] = init_val;
       }
     }
+
+    if (tag == Stack_tag) Stack_sp(result) = 0;
```

Comment:
`caml_alloc` used to initialise `Stack_sp(result)` only in the too-big-for-minor-heap case, which seems like a bug. This commit always initialises `Stack_sp`, in `caml_alloc` for small allocations and in `caml_alloc_shr` for large ones.